### PR TITLE
OSX: Remove tmp .app folder instead of moving to trash

### LINKED ISSUE #42232: Re-export to macOS from Linux throws error in move_to_trash
**Godot version:**

3.2.2.stable.custom_build.bc8eea3e9

**OS/device including version:**

Linux 5.8.8-arch1-1

**Issue description:**

Running "Export All" multiple times can cause an error as Godot fails to overwrite the prior export.

**Steps to reproduce:**

1. Set up an export for Mac (Runnable) `.zip`
2. Run "Export All"
3. Run "Export All" again
4. See error:

```
platform/x11/os_x11.cpp:3438 - move_to_trash: Could not move the resource "/home/rcorre/.cache/godot/JetHook.app" to the trash can "/home/rcorre/.local/share/Trash/files"
```

Manually removing "/home/rcorre/.cache/godot/JetHook.app" avoids the error.

**Minimal reproduction project:**

### PR DESCRIPTION
**Pull request by akien-mga**
Fixes #42232.
And fixes memory leak with use of DirAccess, and harmonize the use of
the sanitized pkg name.

### CODE REVIEW

File: platform/osx/export/export.cpp [hunk 1]
```diff
@@ -574,34 +574,34 @@ Error EditorExportPlatformOSX::export_project(const Ref<EditorExportPreset> &p_p
...
 
-	Error err = OK;
-	String tmp_app_path_name = "";
-
-	DirAccess *tmp_app_path = nullptr;
 	String export_format = use_dmg() && p_path.ends_with("dmg") ? "dmg" : "zip";
 
 	// Create our application bundle.
-	tmp_app_path_name = EditorSettings::get_singleton()->get_cache_dir().plus_file(pkg_name + ".app");
+	String tmp_app_dir_name = pkg_name + ".app";
```
Comment (bruvzg) [inline platform/osx/export/export.cpp:580]: Maybe we should check that "pkg_name" does not have anything nasty, like "../../.." and won't delete anything outside cache.

**Review by bruvzg — APPROVED** [review]

File: platform/osx/export/export.cpp (hunk 1 above)
Comment (akien-mga) [reply platform/osx/export/export.cpp:580]: Should we use `pkg_name_safe` for that which we have a few lines above?

Comment (bruvzg) [reply platform/osx/export/export.cpp:580]: Yes, probably. It should have ".." removed.

Comment (akien-mga) [reply platform/osx/export/export.cpp]: Done. BTW the Mono-specific stuff might still need fixes for ARM64 support as currently it hardcodes `.64` but we'd need to disambiguate the arch.

**Comment by akien-mga** [comment]
I went ahead and made all the logic use the "safe" pkg name for consistency. Might fix potential bugs in filenames.

File: platform/osx/export/export.cpp (hunk 1 above)
Comment (bruvzg) [reply platform/osx/export/export.cpp]: With exception to 3 `dylib`s, Mono related files should be platform independent. And for `dylib`s it's probably better to ship them as universal binaries. In this case all existing export code should work as is (`.64` could be renamed to `.universal` or something like this).

**Review by bruvzg — APPROVED** [review]

File: platform/osx/export/export.cpp [hunk 2]
```diff
@@ -572,36 +572,36 @@ Error EditorExportPlatformOSX::export_project(const Ref<EditorExportPreset> &p_p
...
-	String tmp_app_path_name = "";
+	pkg_name = OS::get_singleton()->get_safe_dir_name(pkg_name);
 
-	DirAccess *tmp_app_path = nullptr;
 	String export_format = use_dmg() && p_path.ends_with("dmg") ? "dmg" : "zip";
 
 	// Create our application bundle.
-	tmp_app_path_name = EditorSettings::get_singleton()->get_cache_dir().plus_file(pkg_name + ".app");
+	String tmp_app_dir_name = pkg_name;
+	String tmp_app_path_name = EditorSettings::get_singleton()->get_cache_dir().plus_file(tmp_app_dir_name);
```
Comment (akien-mga) [inline platform/osx/export/export.cpp:581]: Oops, I introduced a bug here (missing `.app`).

Comment (akien-mga) [reply platform/osx/export/export.cpp:581]: Fix: #44087.

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.4. (Together with #44087.)

**Pull request merged**